### PR TITLE
lightning: use region info to split region

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Optimize region split by skip split regions if there is only one split key and the region size is smaller than region-split-size

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has interface methods change

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

### Release note

 - No release note

### CODE REVIEW

File: pkg/lightning/backend/backend.go [hunk 1]
```diff
@@ -143,7 +146,7 @@ type AbstractBackend interface {
 
 	OpenEngine(ctx context.Context, config *EngineConfig, engineUUID uuid.UUID) error
 
-	CloseEngine(ctx context.Context, engineUUID uuid.UUID) error
+	CloseEngine(ctx context.Context, config *EngineConfig, engineUUID uuid.UUID) error
```
Comment (kennytm) [inline pkg/lightning/backend/backend.go:149]: can the table info be stored in `OpenEngine()`, so we don't need to modify `CloseEngine()`

Comment (glorv) [reply pkg/lightning/backend/backend.go:149]: Not really. When lightning restart at import phase, only CloseEngine is called, in this situation, there is no tableInfo available

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Review by kennytm — APPROVED** [review]

**Pull request merged**